source/fundamentals/crud/compound-operations.txt

```diff
@@ -0,0 +1,313 @@
+   :end-before: end findOneAndReplace-example
+
+The output of the above code should look like this:
+
+.. code-block:: json
+
+   {"_id": 1, "music": "classical", "color": "green"}
+
+For more information on the methods and classes in this section, see the
+following API documentation pages:
+
+- :java-docs:`findOneAndReplace() <apidocs/mongodb-driver-sync/com/mongodb/client/MongoCollection.html#findOneAndReplace(org.bson.conversions.Bson,TDocument,com.mongodb.client.model.FindOneAndReplaceOptions)>`
+- :java-docs:`FindOneAndReplaceOptions <apidocs/mongodb-driver-core/com/mongodb/client/model/FindOneAndReplaceOptions.html>`
+
+Find and Delete
+~~~~~~~~~~~~~~~
+
+To find and delete one document, use the ``findOneAndDelete()`` method of the
+``MongoCollection`` class. The ``findOneAndDelete()`` method returns your found
+document or ``null`` if no documents match your query.
+
+Example
+^^^^^^^
+
+The following example uses the ``findOneAndDelete()`` method to find and
+delete the document with the largest value in the ``_id`` field.
+
+The example uses a ``FindOneAndDeleteOptions`` instance to specify a
+descending sort on the ``_id`` field.
+
+.. literalinclude:: /includes/fundamentals/code-snippets/CompoundOperatorsIndividualExamples.java
+   :language: java
+   :dedent:
+   :start-after: start findOneAndDelete-example
+   :end-before: end findOneAndDelete-example
+
+The output of the above code should look like this:
+
+.. code-block:: json
+
+   {"_id": 2, "food": "pear", "color": "yellow"}
+
+For more information on the ``Sorts`` class, see our
+:doc:`guide on the Sorts builder </fundamentals/builders/sort/>`.
+
+For more information on the methods and classes in this section, see the
+following API documentation pages:
+
+- :java-docs:`findOneAndDelete() <apidocs/mongodb-driver-sync/com/mongodb/client/MongoCollection.html#findOneAndDelete(org.bson.conversions.Bson,com.mongodb.client.model.FindOneAndDeleteOptions)>`
+- :java-docs:`FindOneAndDeleteOptions <apidocs/mongodb-driver-core/com/mongodb/client/model/FindOneAndDeleteOptions.html>`
+
+Avoiding a Race Condition
+-------------------------
+
+In this section we explore two examples. The first example contains a
+race condition, the second example uses a compound operation to
+avoid the race condition present in the first example.
+
+In this section, let's imagine that we run a hotel with one room and that we
+have a small Java program to help us checkout this room to a guest.  
+
+The following document in MongoDB represents the room:
+
+.. code-block:: json
+
+    {"_id": 1, "guest": null, "room": "Blue Room", "reserved": false}
+
+`The full code for this example is available on Github here. <https://github.com/mongodb/docs-java/blob/master/source/includes/fundamentals/code-snippets/CompoundOperators.java>`__
+
+Example With Race Condition
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Let's say our app uses this ``bookARoom`` method to checkout our room to
+a guest:
```

Comment:
```suggestion
Our app uses this ``bookARoom`` method to checkout our room to
a guest:
```
Suggestion: Using "lets say" and "imagine" leaves too much room for imagination. Personally, when I read about an example, I expect concrete scenario's where I shouldn't be wondering about the opposite or tangent scenario.

Applies to all instances.

Reply:
Thanks for the note. As this example is explicitly designed to have you wondering about the opposite scenario (the opposite scenario is presented immediately after this example), so I feel it is appropriate to use open language like "let's say" and "imagine".